src/server/HelixServer.js

```diff
@@ -69,9 +79,13 @@ export class HelixServer extends BaseServer {
       if (liveReload) {
         liveReload.registerFile(ctx.requestId, filePath);
       }
+      log.debug(`${pfx}served local file ${filePath}`);
       return;
     } catch (e) {
-      log.debug(`Error while delivering resource ${ctx.path} - ${e.stack || e}`);
+      log.debug(`${pfx}unable to deliver local file ${ctx.path} - ${e.stack || e}`);
+      if (res.headersSent) {
+        return;
+      }
```

Comment:
this is the actual fix: if serving a local file failed due to a communication error, the response was already initiated but it tried later to serve the proxied resource as well.